test/test.js

```diff
@@ -7,6 +7,14 @@ describe('resolve-dependency-path', function() {
     assert(resolved.indexOf(__dirname) === 0);
   });
 
+  it('resolves with multiple periods in the dependency path', function() {
+    var depPath = './bar.baz.qux';
```

Comment:
Interesting. Thanks for the additional tests. I will look at them. Unfortunately jspm actually does the opposite syntax as requirejs in terms of alternate loaders. It uses 'filename.txt!text' to denote the use of the text loader/plugin. :(

To the original question of the new else block. You are correct, my code broke two out of four of the following cases: './index.js', './index', 'index', and 'index.js'. To handle this, I've added tests for all four of these specific cases since they are very high impact cases. Shortly I will force push a new revision of my code with a modified else case. It should fix the two index.js cases that were previously breaking. Here are the tests I will add.

```js
it('resolves w/initial period, w/ending in .js', function() {
    var depPath = './index';
    var filename = __dirname + '/foo.js';
    var directory = __dirname;
    var resolved = resolvePath(depPath, filename, directory);
    assert(resolved === __dirname + '/index.js');
  });

  it('resolves w/initial period, w/o ending in .js', function() {
    var depPath = './index.js';
    var filename = __dirname + '/foo.js';
    var directory = __dirname;
    var resolved = resolvePath(depPath, filename, directory);
    assert(resolved === __dirname + '/index.js');
  });

  it('resolves w/o initial period, w/o ending in .js', function() {
    var depPath = 'index';
    var filename = __dirname + '/foo.js';
    var directory = __dirname;
    var resolved = resolvePath(depPath, filename, directory);
    assert(resolved === __dirname + '/index.js');
  });

  it('resolves w/o initial period, w/ending in .js', function() {
    var depPath = 'index.js';
    var filename = __dirname + '/foo.js';
    var directory = __dirname;
    var resolved = resolvePath(depPath, filename, directory);
    assert(resolved === __dirname + '/index.js');
  });
```